druid/src/test/java/org/apache/calcite/test/DruidAdapter2IT.java

```diff
@@ -1030,7 +1030,7 @@ private void checkGroupBySingleSortLimit(boolean approx) {
         + "intervals=[[1900-01-09T00:00:00.000Z/2992-01-10T00:00:00.000Z]], "
         + "filter=[AND("
         + "=($3, 'High Top Dried Mushrooms'), "
-        + "SEARCH($87, Sarg['Q2', 'Q3']:CHAR(2)), "
+        + "SEARCH($87, Sarg['Q2':VARCHAR, 'Q3':VARCHAR]:VARCHAR), "
```

Comment:
I find this change surprising.

Reply:
I was surprised too ; however this is the same kind of expressions `SqlToRel` would produce if the input would have been a [set of comparisison](https://github.com/apache/calcite/pull/3835/files#diff-26cbb70deb4731f2c84faf803724dc40a9cdf6e2f2c67f8724e1afbc98fc8950R3226-R3248) ; so the `IN` rewrite now [matches that](https://github.com/apache/calcite/pull/3835/files#diff-26cbb70deb4731f2c84faf803724dc40a9cdf6e2f2c67f8724e1afbc98fc8950R3221)